File: OpenBid/DFP/JavaSample/app/src/main/java/com/pubmatic/openbid/app/MainActivity.java
package com.pubmatic.openbid.app;

import android.Manifest;
import android.content.Context;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.os.Build;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.ActivityCompat;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.DividerItemDecoration;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;

import com.pubmatic.sdk.common.OpenBidSDK;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = "MainActivity";
    private static final String[] PERMISSIONS = {Manifest.permission.ACCESS_FINE_LOCATION,
            Manifest.permission.ACCESS_COARSE_LOCATION,
            Manifest.permission.WRITE_EXTERNAL_STORAGE};

    private static boolean hasPermissions(Context context, String... permissions) {
        if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && context != null && permissions != null) {
            for (String permission : permissions) {
                if (ActivityCompat.checkSelfPermission(context, permission) != PackageManager.PERMISSION_GRANTED) {
                    return false;
                }
            }
        }
        return true;
    }

    private  void initListView() {

        RecyclerView adList =  findViewById(R.id.ad_list);
        adList.setHasFixedSize(true);
        adList.addItemDecoration(new DividerItemDecoration(this, DividerItemDecoration.VERTICAL));

        final List<AD_TYPE> itemList = new ArrayList<>(Arrays.asList(AD_TYPE.values()));

        RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(this);
        adList.setLayoutManager(layoutManager);

        AdListAdapter adListAdapter = new AdListAdapter(itemList);
        adListAdapter.setItemClickListener(new AdListAdapter.OnItemClickListener() {
            @Override
            public void onClick(View view, int position) {
                Class activityClass = itemList.get(position).getActivity();
                if (activityClass != null) {
                    loadAdScreen(activityClass);
                }
            }
        });
        adList.setAdapter(adListAdapter);

    }
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        if(BuildConfig.DEBUG){
            OpenBidSDK.setLogLevel(OpenBidSDK.LogLevel.Debug);
        }

        //Change the Status Bar color
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            Window window = getWindow();
            window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
            window.setStatusBarColor(getResources().getColor(R.color.colorStatusBar));
        }

        // Ask permission from user for location and write external storage
        if (!hasPermissions(MainActivity.this, PERMISSIONS)) {
            int MULTIPLE_PERMISSIONS_REQUEST_CODE = 123;
            ActivityCompat.requestPermissions(MainActivity.this, PERMISSIONS, MULTIPLE_PERMISSIONS_REQUEST_CODE);
        }

        initListView();
    }

    public void loadAdScreen(@NonNull Class activity) {
        Intent intent = new Intent(MainActivity.this, activity);
        startActivity(intent);
    }

    enum AD_TYPE {
        BANNER(DFPBannerActivity.class, "Banner"),
        INTERSTITIAL(DFPInterstitialActivity.class, "Interstitial"),
        VIDEO_INTERSTITIAL(DFPVideoInterstitialActivity.class, "Video Interstitial");

        private Class activity;
        private String displayText;

        AD_TYPE(@Nullable Class activity, @NonNull String text) {
            this.activity = activity;
            this.displayText = text;
        }

        public Class getActivity() {
            return activity;
        }

        public String getDisplayText() {
            return displayText;
        }
    }
}

File: OpenBid/DFP/KotlinSample/app/src/main/java/com/pubmatic/openbid/kotlinsampleapp/MainActivity.kt
package com.pubmatic.openbid.kotlinsampleapp

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView

import kotlinx.android.synthetic.main.activity_home.*


class MainActivity : AppCompatActivity()  {

    var recycler: RecyclerView ? = null
    var list: ArrayList<String> ? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_home)
        setSupportActionBar(toolbar)

        recycler = findViewById(R.id.recycler_view)
        recycler!!.layoutManager = LinearLayoutManager(this)
        recycler!!.setHasFixedSize(true)

        list = ArrayList<String>()
        list?.add("Banner")
        list?.add("Interstitial")
        val recyclerAdapter = RecyclerAdapter(list, RecyclerItemListener())
        recycler!!.adapter = recyclerAdapter
    }

    fun displayActivity(position: Int){
        val intent :Intent
        if(position == 0){
            intent = Intent(this, DFPBannerActivity::class.java)
        }else{
            intent = Intent(this, DFPInterstitialActivity::class.java)
        }
        startActivity(intent)
    }


    inner class RecyclerItemListener : RecyclerAdapter.OnItemClickListener{
        override fun onItemClick(position: Int) {
            displayActivity(position)
        }

    }







}
